### DISCUSSION TITLE
TSL2591 Light Sensor - return NAN value

### DISCUSSION THREAD

**Discussion by besbar** [UncleRus/esp-idf-lib#518] [Q&A]
Dear community, 

Thank you so much for this repository. I've recenty started using ESP-IDF (in order to exploit Bluetooth mesh techno.).
I am using ESP-IDF v5.01.
I wanted to make a BT mesh multi-sensors with so multiple sensors.

Amongst I2C devices, I made working together on a ESP32 Adafruit Feather following sensors:
- SHT31 or AHT20 (°C and %RH),
- MCP9808 (°C)
- SGP30 (TVOC/eCO2),
- OLED Screen SH1107,
- LC709203F (battery monitor),   
- BH1750 or VEML7700 (light sensors)

However I meet difficulties with the TSL2591 light sensor, when I use the component (exactly the same way as for sensors above), results returned are nan Lux.
I am using the TSL2591 ADA1980 from Adafruit.
Is there anybody here that has made this sensor working properly?

It is worth noting that this sensor is working in Arduino IDE with adafruit library.

Thanks a lot for any help on that topic.

**Comment by UncleRus** [comment]
It looks like a bug in library.
I'm transferring this discussion into an issue
